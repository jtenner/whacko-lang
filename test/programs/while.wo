export fn _start(): void {
  const cond = true;
  while (cond) {
    let a = 0;
    while (a < 10) {
      a++;
      continue;
    }

    break;
  }
}