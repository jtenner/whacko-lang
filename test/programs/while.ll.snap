; ModuleID = 'whacko'
source_filename = "whacko"
target triple = "wasm32-wasi"

define void @_start() {
entry:
  %"alloca~0" = alloca i1, align 1
  %"alloca~1" = alloca i64, align 8
  br label %"condition~2"

"condition~2":                                    ; preds = %"unused~24", %entry
  %"4~inst" = load i1, ptr %"alloca~0", align 1
  br i1 %"4~inst", label %"body~5", label %"next~6"

"body~5":                                         ; preds = %"condition~2"
  br label %"condition~8"

"next~6":                                         ; preds = %"next~14", %"condition~2"
  ret void

"condition~8":                                    ; preds = %"unused~21", %"body~13", %"body~5"
  %"10~inst" = load i64, ptr %"alloca~1", align 4
  %"12~inst" = icmp slt i64 %"10~inst", 10
  br i1 %"12~inst", label %"body~13", label %"next~14"

"body~13":                                        ; preds = %"condition~8"
  %"16~inst" = load i64, ptr %"alloca~1", align 4
  %"17~inst" = load i64, ptr %"alloca~1", align 4
  %"19~inst" = add i64 %"17~inst", 1
  store i64 %"19~inst", ptr %"alloca~1", align 4
  br label %"condition~8"

"next~14":                                        ; preds = %"condition~8"
  br label %"next~6"

"unused~21":                                      ; No predecessors!
  br label %"condition~8"

"unused~24":                                      ; No predecessors!
  br label %"condition~2"
}
